### PR TITLE
Pesto flutter steps

### PR DESCRIPTION
**Pull request by domesticmouse**
Turning `pesto_flutter` into a series of steps

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: pesto_flutter/final/macos/Runner/DebugProfile.entitlements [hunk 1]
```diff
@@ -6,8 +6,6 @@
 	<true/>
 	<key>com.apple.security.cs.allow-jit</key>
 	<true/>
-	<key>com.apple.security.network.client</key>
```
Comment (domesticmouse) [inline pesto_flutter/final/macos/Runner/DebugProfile.entitlements:9]: Any particular reason why the macOS app requires network client entitlement?

File: pesto_flutter/final/macos/Runner/Release.entitlements [hunk 2]
```diff
@@ -4,9 +4,5 @@
 <dict>
 	<key>com.apple.security.app-sandbox</key>
 	<true/>
-	<key>com.apple.security.network.client</key>
```
Comment (domesticmouse) [inline pesto_flutter/final/macos/Runner/Release.entitlements:7]: Likewise, the release entitlments. 

PTAL @jwill

File: pesto_flutter/final/macos/Runner/DebugProfile.entitlements (hunk 1 above)
Comment (jwill) [reply pesto_flutter/final/macos/Runner/DebugProfile.entitlements:9]: This is a default addition from generation or perhaps adding packages. This wasn't added or selected by hand.

File: pesto_flutter/final/macos/Runner/Release.entitlements (hunk 2 above)
Comment (jwill) [reply pesto_flutter/final/macos/Runner/Release.entitlements:7]: This is a default addition from generation or perhaps adding packages. This wasn't added or selected by hand.

File: pesto_flutter/final/macos/Runner/DebugProfile.entitlements (hunk 1 above)
Comment (domesticmouse) [reply pesto_flutter/final/macos/Runner/DebugProfile.entitlements:9]: Weird. The delta here comes from re-generating all of the runner packages (`android`, `ios`, `linux`, `macos`, `windows`). I'm unaware of any packages that will edit these files on installation.

File: pesto_flutter/final/macos/Runner/Release.entitlements (hunk 2 above)
Comment (domesticmouse) [reply pesto_flutter/final/macos/Runner/Release.entitlements:7]: understood

**Pull request closed without merging**